designs/v1-roadmap.md

```diff
@@ -0,0 +1,270 @@
+
+#### Tasks
+
+- [ ] Remove the knative logger and replace with the controller-runtime logger
+- [ ] Update the status condition schema for Karpenter CustomResources to use the [metav1 status condition schema](https://github.com/kubernetes/apimachinery/blob/f14778da5523847e4c07346e3161a4b4f6c9186e/pkg/apis/meta/v1/types.go#L1523)
+
+### Drop Knative Webhook from Karpenter
+
+**Issue Ref:** https://github.com/kubernetes-sigs/karpenter/issues/332
+
+**Category:** Stability
+
+As part of Karpenter completely removing its dependency on Knative, Karpenter needs to remove its coupling on the knative webhook certificate reconciliation logic. Currently, Karpenter leverages knative’s webhook reconciler to reconcile a certificate needed to enable the TLS webhook traffic. As part of removing this dependency that `kubernetes-sigs/karpenter` has on the knative webhook reconciliation logic, Karpenter will be dropping webhooks entirely.
+
+ Because K8s versions 1.25+ have support for CustomResourceValidations driven through Common Expression Language, Karpenter is dropping support for the webhook container entirely.
+
+[`Azure/karpenter-provider-azure`](https://github.com/Azure/karpenter) dropped support for Kubernetes 1.25 since AKS [dropped support for this version in 2023.](https://learn.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli#aks-kubernetes-release-calendar) [`aws/karpenter-provider-aws`](https://github.com/aws/karpenter-provider-aws) still supports versions [all the way back to Kubernetes 1.23 due to extended support](https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html). The AWS provider will continue to support versions prior to 1.25 until January 31, 2025. At this point, if not other providers support versions prior to 1.25, we will drop support for webhooks in [`kubernetes-sigs/karpenter`](https://github.com/kubernetes-sigs/karpenter).
+
+#### Tasks
+
+- [ ] Drop the webhook, removing the `knative/pkg` dependency from the Karpenter controller package
+
+### Promoting Drift Feature to Stable
+
+**Category:** Stability
+
+Karpenter supported drift in its alpha state from v0.21-v0.32. During alpha, we worked to build out features and promoted drift to beta after releasing full drift support for all Karpenter configuration. Since releasing drift in beta, we’ve received no feedback that would lead us to believe the feature is unstable or not the right direction for the declarative state of Karpenter.
+
+Since the feature is such a fundamental part to how the declarative state of Karpenter functions, we will promote drift to stable at Karpenter v1.
+
+#### Tasks
+
+- [ ] Remove `Drift` as a feature flag, always enabling it in Karpenter
+
+### Change default TopologySpreadConstraint policy for Deployment from `ScheduleAnyways` to `DoNotSchedule` [AWS CloudProvider]
```

Comment:
Should we consider making an RFC for AWS provider in that repo? I'm not sure this makes much sense in the k8s-sigs context